packages/migrate/migrations/routes/migrate_page_js/index.js

```diff
@@ -24,21 +24,21 @@ export function migrate_page(content) {
 	const file = parse(content);
 	if (!file) return give_up + content;
 
-	if (!file.exports.map.has('load') && !file.exports.complex) {
+	const name = file.exports.map.get('load');
+
+	if (!name && !file.exports.complex) {
 		// there's no load function here, so there's nothing to do
 		return content;
 	}
 
-	const name = file.exports.map.get('load');
-
 	for (const statement of file.ast.statements) {
-		const fn = get_function_node(statement, name);
-		if (fn) {
+		const fn = name ? get_function_node(statement, name) : undefined;
+		if (fn?.body) {
```

Comment:
not sure i understand this — when would a function node not have a body? do we need to tighten up the implementation of `get_function_node`?

Reply:
I don't think we can do much here, that's typescript giving it an optional type. I don't think this can happen except when you have a syntax error, and unfortunately we can't just use `!` 😄

Reply:
ah, of course — no requirement for the AST to be well formed. error-tolerant parsers are new to me!